controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -152,125 +164,287 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		// We have a match that isn't running and can invoke the job
 		r.Log.Info("Invoking job", "jobinvocation", jobName, "namespace", app.Namespace)
 
-		if err := r.InvokeJob(ctx, &job, &app, &env, &cji); err != nil {
-			r.Log.Info("Job Invocation Failed", "jobinvocation", jobName, "namespace", app.Namespace)
+		if err := r.InvokeJob(&cache, &job, &app, &env, &cji); err != nil {
+			r.Log.Error(err, "Job Invocation Failed", "jobinvocation", jobName, "namespace", app.Namespace)
 			r.Recorder.Eventf(&cji, "Warning", "JobNotInvoked", "Job [%s] could not be invoked", jobName)
 			return ctrl.Result{Requeue: true}, err
 		}
 	}
 
+	// Check IQE struct to see if we need to invoke an IQE Job
+	// In the future, we'll need ot handle other types, but this will suffice since testing only has iqe.
+	var emptyTesting crd.IqeJobSpec
+	if cji.Spec.Testing.Iqe != emptyTesting {
+
+		nn := types.NamespacedName{
+			Name:      fmt.Sprintf("%s-iqe", cji.Name),
+			Namespace: cji.Namespace,
+		}
+
+		j := batchv1.Job{}
+
+		if err := r.createIqeJobResource(&cache, &cji, &env, &app, nn, ctx, &j); err != nil {
+			r.Log.Error(err, "Iqe Job creation encountered an error", "jobinvocation", nn.Name)
+			r.Recorder.Eventf(&cji, "Warning", "IQEJobFailure", "Job [%s] failed to invoke", j.ObjectMeta.Name)
+			return ctrl.Result{Requeue: true}, err
+		}
+		if err := cache.Create(IqeClowdJob, nn, &j); err != nil {
+			r.Log.Error(err, "Iqe Job could not update via cache", "jobinvocation", nn.Name)
+			r.Recorder.Eventf(&cji, "Warning", "IQECacheUpdateFailed", "Job [%s] could not update", j.ObjectMeta.Name)
```

Comment:
Personally I think reporting the Cache as a top level issue is a little too granular.....IQEJobFailure is fine, with a message about not being able to update the cache.